tests/test_vqe.py

```diff
@@ -184,6 +184,23 @@ def test_hamiltonian_valid_init(self, coeffs, ops):
         H = qml.vqe.Hamiltonian(coeffs, ops)
         assert H.terms == (coeffs, ops)
 
+    @pytest.mark.parametrize(
+        ("coeffs", "ops", "is_diagonal"),
+        [
+            ([1, 1, 1], [qml.PauliZ(0), qml.PauliZ(1), qml.PauliZ(2)], True),
+            ([1, 1, 1], [qml.PauliZ(0) @ qml.PauliZ(1), qml.PauliZ(1), qml.PauliZ(2)], True),
+            ([1, 1, 1], [qml.PauliZ(0), qml.PauliZ(1), qml.PauliX(2)], False),
+            ([1, -1, 1], [qml.PauliX(0) @ qml.PauliX(1), qml.PauliX(0) @ qml.PauliX(1), qml.Identity(2)], True),
+            ([1, -1, 2], [qml.PauliX(0) @ qml.PauliZ(1), qml.PauliX(0) @ qml.PauliZ(1) @ qml.Identity(2), qml.Identity(2)], True),
+            ([1, -1, 2], [qml.Hermitian(np.array([[0, 1], [1, 0]]), 0) @ qml.Identity(1), qml.PauliX(0), qml.PauliZ(1)], True)
```

Comment:
Would also be good to include a test case with `qml.Hermitian` that is diagonal!